File: php/src/Creational/AbstractFactory/HonoluluConcreteSoupFactory.php
<?

#//HonoluluConcreteSoupFactory.java - One of Two concrete factories extending the abstract factory

require_once 'AbstractSoupFactory.php';


class HonoluluConcreteSoupFactory extends AbstractSoupFactory
{
	function __construct()
	{
			$this->factoryLocation = "Honolulu";
	}

	function makeClamChowder() 
	{
		return new HonoluluClamChowder();
	}

	function makeFishChowder() 
	{
		return new HonoluluFishChowder();
	}
}

File: go/creational/abstractfactory/HonoluluConcreteSoupFactory.go
package abstractfactory

import "../common"

type HonoluluConcreteSoupFactory struct {
	BaseSoupFactory
}

func NewHonoluluConcreteSoupFactory () *HonoluluConcreteSoupFactory {
	return &HonoluluConcreteSoupFactory{ BaseSoupFactory{ "Honolulu" } }
}

func (this *HonoluluConcreteSoupFactory) MakeClamChowder() common.Soup {
	return common.NewHonoluluClamChowder()
}

func (this *HonoluluConcreteSoupFactory) MakeFishChowder() common.Soup {
	return common.NewHonoluluFishChowder()
}

